.test-infra/jenkins/job_LoadTests_Combine_Java.groovy

```diff
@@ -0,0 +1,216 @@
+                            maxNumWorkers       : 5,
+                            numWorkers          : 5,
+                            autoscalingAlgorithm: "NONE",
+                            perKeyCombiner      : "TOP_LARGEST",
+                            streaming           : streaming
+                    ]
+            ],
+            [
+                    title        : 'Load test: fanout 8 times with 2GB 10-byte records total',
+                    itClass      : 'org.apache.beam.sdk.loadtests.CombineLoadTest',
+                    runner       : CommonTestProperties.Runner.DATAFLOW,
+                    jobProperties: [
+                            project             : 'apache-beam-testing',
+                            appName             : "load_tests_Java_Dataflow_${jobName}_Combine_5",
+                            tempLocation        : 'gs://temp-storage-for-perf-tests/loadtests',
+                            publishToBigQuery   : true,
+                            bigQueryDataset     : 'load_test',
+                            bigQueryTable       : "java_dataflow_${jobName}_Combine_5",
+                            sourceOptions       : """
+                                                    {
+                                                      "numRecords": 5000000,
+                                                      "keySizeBytes": 10,
+                                                      "valueSizeBytes": 90
+                                                    }
+                                               """.trim().replaceAll("\\s", ""),
+                            fanout              : 8,
+                            iterations          : 1,
+                            topCount            : 20,
+                            maxNumWorkers       : 5,
+                            numWorkers          : 5,
+                            autoscalingAlgorithm: "NONE",
+                            perKeyCombiner      : "TOP_LARGEST",
+                            streaming           : streaming
+                    ]
+            ]
+    ]
+}
+
+
+def batchLoadTestJob = { scope, triggeringContext ->
+    scope.description('Runs Java Combine load tests on Dataflow runner in batch mode')
+    commonJobProperties.setTopLevelMainJobProperties(scope, 'master', 240)
+
+    for (testConfiguration in commonLoadTestConfig('batch', false)) {
+        loadTestsBuilder.loadTest(scope, testConfiguration.title, testConfiguration.runner, CommonTestProperties.SDK.JAVA, testConfiguration.jobProperties, testConfiguration.itClass, triggeringContext)
+    }
+}
+
+def streamingLoadTestJob = {scope, triggeringContext ->
+    scope.description('Runs Java Combine load tests on Dataflow runner in streaming mode')
+    commonJobProperties.setTopLevelMainJobProperties(scope, 'master', 240)
+
+    for (testConfiguration in commonLoadTestConfig('combine', true)) {
+        testConfiguration.jobProperties << [inputWindowDurationSec: 1200]
```

Comment:
Maybe it would be better to have it in jobProperties above to keep everything all together.

Reply:
I'm reluctant to put a parameter connected only with streaming in a common configuration. I don't know a way of putting a whole key-value pair in a map conditionally